tidypolars/tibble.py

```diff
@@ -949,6 +960,110 @@ def write_parquet(self,
         """Write a data frame to a parquet"""
         return super().write_parquet(file, compression = compression, use_pyarrow = use_pyarrow, **kwargs)
 
+    def group_by(self, group, *args, **kwargs):
+        res = TibbleGroupBy(self, group, maintain_order=True)
+        return res
+
+    def nest(self, by, select='all', nested_name="data", *args, **kwargs):
+        """
+        Nest rows into a list-column of dataframes
+
+        Parameters
+        ----------
+        by : list, str
+            Columns to nest on
+        select : str, list
+            Columns to select for the nested dataframe. If 'all' (default)
+            all columns except those specified in 'by' will be selected.
+        nested_name : str
+            Name of the column to receive the nested dataframe
+
+        Examples
+        --------
+        """
+        if select=='all':
+            select = [col for col in self.names if col not in by]
+        # make sure all columns in 'by' are removed from the nested data
+        select = [col for col in select if col not in by]
+        out = (self
+               .group_by(by)
+               .agg(**{
+                   nested_name : pl.struct(select).map_elements(
+                       lambda cols: from_polars( pl.DataFrame(cols.to_list()) ) )
+                   })
+               )
+        return out.pipe(from_polars)
+    
+    def unnest(self, col, *args, **kwargs):
+        """
+        Unnest a nested data frame
+        Parameters
+        ----------
+        col : str
+            Columns to unnest
+
+        """
+        assert isinstance(col, str), "'col', must be a string"
+        out = (self
+               .mutate(**{
+                   col : pl.col(col).map_elements(lambda d: d.to_struct())
+               })
+               .to_polars()
+               .explode(col)
+               .unnest(col)
+               )
+        return out.pipe(from_polars)
+
+    def crossing(self, *args, **kwargs):
+        """
+        Expand the data set using a list of values. Each value in the
+        list
+        """
+        out = self.mutate(*args, **kwargs).to_polars()
+        for var,_ in kwargs.items():
+            out = out.explode(var)
+        return out.pipe(from_polars)
+
+    # Not tidy functions, but useful from pandas/polars 
+    # -------------------------------------------------
+    def replace(self, *args, **kwargs):
```

Comment:
`.replace()` should be removed for now. It might be worth adding in the future but not now. We would also want to write this in polars syntax - I'm guessing `pandas` is relatively slow on larger datasets.

Reply:
Well, why tie our hands and not use some methods that are better in Python than in R, like replace? `pandas` is slower, but the method is still very useful.

Reply:
Hmm I might be convinced of this in the future - as of right now this functionality is covered enough using `tp.if_else()`

Reply:
Replace is much more general to replace strings in a column, which can be used with or without regular expression. It is a shame that tidyverse does not have something like this.

Reply:
Wouldn't this go to `.mutate(str_replace())` then?

Reply:
The advantage of replace is that it can use a dictionary to do the replacements, and we can specify all the columns to do the replacements at once. Polars do not have labelled data. replace() makes things much easier to replace values in a labelled-like way when the labels are stored in a dictionary, like in SAS, DTA and SAV files, when we need to switch between numbers and their label.